packages/common-utils/src/types.ts

```diff
@@ -341,9 +344,24 @@ export const _ChartConfigSchema = z.object({
   metricTables: MetricTableSchema.optional(),
 });
 
+export const CteExpressions = z.object({
+  with: z
+    .array(
+      z.object({
+        name: z.string(),
+        sql: z
+          .string()
+          .or(SelectSQLStatementSchema)
+          .or(_ChartConfigSchema)
+          .or(z.custom<ChSql>()),
```

Comment:
does this do any validation? I suspect we want to infer ChSql from the zod schema instead. Like
```
const ChSqlSchema = z.object({
  sql: z.string(),
  params: z.record(z.any()),
});

export type ChSql = z.infer<typeof ChSqlSchema>;
```

Reply:
Addressed in PR. There's now a `ChSqlSchema` defined for this instead of using `z.custom`.